python/cuml/cuml/manifold/umap.pyx

```diff
@@ -430,6 +440,42 @@ class UMAP(UniversalBase,
 
         self.build_kwds = build_kwds
 
+        self._handle_deprecated_build_kwds()
+
+    # TODO: remove deprecation handling logic in 25.08
+    # related issue: https://github.com/rapidsai/cuml/issues/6742
+    def _handle_deprecated_build_kwds(self):
```

Comment:
This should be done at the site of usage of build build_kwds (within `_build_umap_params`), not in call within `__init__` alone. sklearn compatible APIs expect to be able to change input parameters later via `set_params` or `__setattr__`, handling this in `__init__` only (and via mutating a user-provided dict) is insufficient and may result in subtle bugs. See https://scikit-learn.org/stable/developers/develop.html#instantiation for more info.

I'd recommend modifying `_build_umap_params` to handle both kinds of input (the new and the old) _without_ mutating the `.build_kwds` mapping in the process.

Reply:
@jcrist FYI, I had pointed out to @jinsolp offline that we should avoid modifying or processing hyperparameters as part of the estimator constructor, however recommended to not fix (all of this) as part of this PR considering that we are already violating this rule, e.g., for the `build_algo` parameter (and I assume some others). It's probably a good idea to not introduce even more errors along those lines, so I agree with your request, but I don't want this PR to be blocked on this.

Reply:
It shouldn't be that hard to do correctly though as part of `_build_umap_params`. Agree though - I'm more blocking on ensuring if we're making deprecating changes to the kwargs that we're ending up in a better place (and won't be deprecating again next release).